File: Catalog_Service/src/main/java/cf/catalog_service/srevices/StoreService.java
package cf.catalog_service.srevices;

import cf.catalog_service.dto.store.StoreRequestDto;
import cf.catalog_service.dto.store.StoreResponseDto;
import cf.catalog_service.enums.MainCategory;

import java.util.List;
import java.util.Optional;

public interface StoreService {
    StoreResponseDto createStore(StoreRequestDto requestDto);
    StoreResponseDto getStoreById(String id);
    List<StoreResponseDto> getAllStores();
    StoreResponseDto updateStore(String id, StoreRequestDto requestDto);
    List<StoreResponseDto> getStoreByName(String name);
    void deleteStore(String id);

    // Recherches spécifiques
    List<StoreResponseDto> getStoresByOwner(String ownerId);
    List<StoreResponseDto> getStoresByCategory(MainCategory category);
    List<StoreResponseDto> getOpenStores();
}

File: simovite-frontend/src/app/services/store.service.ts
// store.service.ts

import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment'; // Adapte le chemin
import { StoreRequestDto, StoreResponseDto } from '../models/store.model'; // Adapte le chemin
import { MainCategory } from '../models/store.model'; // Adapte le chemin
import { fr } from 'date-fns/locale';
@Injectable({
  providedIn: 'root'
})
export class StoreService {

  private readonly apiUrl = `${environment.apiGateway}/CATALOG-SERVICE/v1/stores`;

  constructor(private http: HttpClient) { }

  // ==========================================
  // CRUD BASIQUE
  // ==========================================

  createStore(requestDto: StoreRequestDto): Observable<StoreResponseDto> {
    return this.http.post<StoreResponseDto>(this.apiUrl, requestDto);
  }

  getStoreById(id: string): Observable<StoreResponseDto> {
    return this.http.get<StoreResponseDto>(`${this.apiUrl}/${id}`);
  }

  getAllStores(): Observable<StoreResponseDto[]> {
    return this.http.get<StoreResponseDto[]>(this.apiUrl);
  }

  updateStore(id: string, requestDto: StoreRequestDto): Observable<StoreResponseDto> {
    return this.http.put<StoreResponseDto>(`${this.apiUrl}/${id}`, requestDto);
  }

  deleteStore(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  // ==========================================
  // RECHERCHES SPÉCIFIQUES
  // ==========================================

  getStoreByName(name: string): Observable<StoreResponseDto[]> {
    let params = new HttpParams().set('name', name);
    return this.http.get<StoreResponseDto[]>(`${this.apiUrl}/search`, { params });
  }

  getStoresByOwner(ownerId: string): Observable<StoreResponseDto[]> {
    return this.http.get<StoreResponseDto[]>(`${this.apiUrl}/owner/${ownerId}`);
  }

  getStoresByCategory(category: MainCategory): Observable<StoreResponseDto[]> {
    return this.http.get<StoreResponseDto[]>(`${this.apiUrl}/category/${category}`);
  }

  getOpenStores(): Observable<StoreResponseDto[]> {
    return this.http.get<StoreResponseDto[]>(`${this.apiUrl}/open`);
  }
}